File: 03_FizzBuzz_Reformating/FizzBuzzConverter.java
class FizzBuzzConverter {

    public static void printFizzBuzz() {
        int n = 100;

        for (int i = 1; i <= n; i++) {

            if (i % 3 == 0 && i % 5 == 0) {
                isFizzBuzz();
            } else if (i % 3 == 0) {
                isFizz();
            } else if (i % 5 == 0) {
                isFuzz();
            } else {
                System.out.print(i);
            }

            System.out.print(", " );
        }

    }

    private static void isFizzBuzz() {
        System.out.print("FizzBuzz");
    }

    private static void isFizz() {
        System.out.print("Fizz");
    }

    private static void isFuzz() {
        System.out.print("Buzz");
    }

}

File: FizzBuzzChallenge/src/main/kotlin/ch/adesso/educamp/fizzbuzz/FizzBuzzConverter.kt
package ch.adesso.educamp.fizzbuzz

class FizzBuzzConverter {
    companion object {
        private fun isFizz(number: Int): Boolean {
            return number % 3 == 0
        }

        private fun isBuzz(number: Int): Boolean {
            return number % 5 == 0
        }

        fun isFizzBuzz(number: Int): String {
            val isFizz = isFizz(number)
            val isBuzz = isBuzz(number)
            return if (isFizz && isBuzz) {
                FIZZ + BUZZ
            } else if (isFizz) {
                FIZZ
            } else if (isBuzz) {
                BUZZ
            } else {
                number.toString()
            }
        }

        private const val FIZZ: String = "Fizz"
        private const val BUZZ: String = "Buzz"
    }
}
